core/src/test/java/org/infinispan/notifications/cachelistener/CacheNotifierImplWithConverterTest.java

```diff
@@ -0,0 +1,322 @@
+      assert ((CacheEntryActivatedEvent) cl.getEvents().get(1)).getValue().equals("custom event for test (CACHE_ENTRY_ACTIVATED,k,v)");
+   }
+
+   public void testNotifyCacheEntryPassivated() {
+      n.notifyCacheEntryPassivated("k", "v", true, null, null);
+      n.notifyCacheEntryPassivated("k", "v", false, null, null);
+
+      assert cl.isReceivedPost();
+      assert cl.isReceivedPre();
+      assert cl.getInvocationCount() == 2;
+      assert cl.getEvents().get(0).getCache() == mockCache;
+      assert cl.getEvents().get(0).getType() == Event.Type.CACHE_ENTRY_PASSIVATED;
+      CacheEntryPassivatedEvent event = (CacheEntryPassivatedEvent) cl.getEvents().get(0);
+      assert event.getKey().equals("k");
+      assert event.getValue().equals("custom event for test (CACHE_ENTRY_PASSIVATED,k,v)");
+      assert cl.getEvents().get(1).getCache() == mockCache;
+      assert cl.getEvents().get(1).getType() == Event.Type.CACHE_ENTRY_PASSIVATED;
+      event = (CacheEntryPassivatedEvent) cl.getEvents().get(1);
+      assert event.getKey().equals("k");
+      assert event.getValue().equals("custom event for test (CACHE_ENTRY_PASSIVATED,k,v)");
+   }
+
+   public void testNotifyTransactionCompleted() {
+      GlobalTransaction tx = mock(GlobalTransaction.class);
+      n.notifyTransactionCompleted(tx, true, ctx);
+      n.notifyTransactionCompleted(tx, false, ctx);
+
+      assert cl.getInvocationCount() == 2;
+      assert cl.getEvents().get(0).getCache() == mockCache;
+      assert cl.getEvents().get(0).getType() == Event.Type.TRANSACTION_COMPLETED;
+      assert ((TransactionCompletedEvent) cl.getEvents().get(0)).isTransactionSuccessful();
+      assert ((TransactionCompletedEvent) cl.getEvents().get(0)).getGlobalTransaction() == tx;
+      assert cl.getEvents().get(1).getCache() == mockCache;
+      assert cl.getEvents().get(1).getType() == Event.Type.TRANSACTION_COMPLETED;
+      assert !((TransactionCompletedEvent) cl.getEvents().get(1)).isTransactionSuccessful();
+      assert ((TransactionCompletedEvent) cl.getEvents().get(1)).getGlobalTransaction() == tx;
+   }
+
+   public void testNotifyTransactionRegistered() {
+      GlobalTransaction tx = mock(GlobalTransaction.class);
+      n.notifyTransactionRegistered(tx, false);
+      n.notifyTransactionRegistered(tx, false);
+
+      assert cl.getInvocationCount() == 2;
+      assert cl.getEvents().get(0).getCache() == mockCache;
+      assert cl.getEvents().get(0).getType() == Event.Type.TRANSACTION_REGISTERED;
+      assert ((TransactionRegisteredEvent) cl.getEvents().get(0)).getGlobalTransaction() == tx;
+      assert cl.getEvents().get(1).getCache() == mockCache;
+      assert cl.getEvents().get(1).getType() == Event.Type.TRANSACTION_REGISTERED;
+      assert ((TransactionRegisteredEvent) cl.getEvents().get(1)).getGlobalTransaction() == tx;
+   }
+   
+   @NamedFactory(name = "to-string-converter-factory")
+   public class ToStringEventConverterFactory implements CacheEventConverterFactory {
```

Comment:
We don't need the factory, just having the `ToStringConverter` class is more than enough.